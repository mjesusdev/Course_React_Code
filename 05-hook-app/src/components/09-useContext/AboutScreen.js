import React, { useContext } from 'react'
import { UserContext } from './UserContext'

export const AboutScreen = () => {

    const { user, setUser } = useContext( UserContext );

    const handleClick = () => {
        setUser({});
    }

    return (
        <div>
            <h1>AboutScreen</h1>
            <hr />

            <pre>
                { JSON.stringify( user, null, 3 ) }
            </pre>


            <button 
                className="btn btn-warning"
                onClick={ handleClick }
            >
                Logout
            </button>

        </div>
    )
}
