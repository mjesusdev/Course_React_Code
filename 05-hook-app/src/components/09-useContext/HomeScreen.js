import React, { useContext } from 'react'
import { UserContext } from './UserContext';

export const HomeScreen = () => {

    const { user } = useContext(UserContext);

    // console.log( user );

    return (
        <div>
            <h1>HomeScreen</h1>
            <hr />

            <pre>
                { JSON.stringify( user, null, 3 ) }
            </pre>

        </div>
    )
}
