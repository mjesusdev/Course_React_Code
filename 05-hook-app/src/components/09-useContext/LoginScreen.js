import React, { useContext } from 'react';
import { UserContext } from './UserContext';

export const LoginScreen = () => {

    const { setUser } = useContext( UserContext );

    return (
        <div>
            <h1>LoginScreen</h1>
            <hr />
            <button 
                className="btn btn-primary"
                onClick={ ()=> setUser({
                    id: 123,
                    name: 'Fernando'
                })}
            >
                Login
            </button>
        </div>
    )
}
