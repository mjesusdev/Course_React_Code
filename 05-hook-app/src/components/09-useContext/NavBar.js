import React from 'react';
import { Link, NavLink } from 'react-router-dom';

export const NavBar = () => {
    return (
        <nav className="navbar navbar-expand-sm navbar-dark bg-dark">

            <Link to="/" className="navbar-brand"> useContext </Link>

            <div className="collapse navbar-collapse" id="navbarNavAltMarkup">
                <div className="navbar-nav">
                    <NavLink exact activeClassName="active" to="/" className="nav-item nav-link"> Home </NavLink>
                    <NavLink exact activeClassName="active" to="/about" className="nav-item nav-link"> About </NavLink>
                    <NavLink exact activeClassName="active" to="/login" className="nav-item nav-link"> Login</NavLink>
                </div>
            </div>
        </nav>
    )
}
