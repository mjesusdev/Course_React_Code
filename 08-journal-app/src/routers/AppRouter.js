import React from 'react';
import {
    BrowserRouter as Router,
    Redirect,
    Route,
    Switch
} from 'react-router-dom';

import { JournalScreen } from '../components/journal/JournalScreen';
import { AuthRouter } from './AuthRouter';

export const AppRouter = () => {
    return (
        <Router>
            <div>
                <Switch> 
                    <Route  
                        path="/auth" 
                        component={ AuthRouter } 
                    />
                    
                    <Route 
                        exact
                        path="/"
                        component={ JournalScreen }
                    />

                    <Redirect to="/auth/login" />
                    
                </Switch>
            </div>
        </Router>
    )
}
