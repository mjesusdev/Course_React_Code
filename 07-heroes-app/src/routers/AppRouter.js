import React, { useContext } from 'react'
import { Routes } from 'react-router-dom';

import { AuthContext } from '../auth/AuthContext';
import { PrivateRoute } from './PrivateRoute';
import { PublicRoute } from './PublicRoute';

import { LoginScreen } from '../components/login/LoginScreen';
import { DashboardRoutes } from './DashboardRoutes';

export const AppRouter = () => {

    const { user } = useContext(AuthContext);

    return (
        <div>
            <Routes> 
                <PublicRoute
                    path="/login"
                    element={ <LoginScreen /> } 
                    isAutenticated={ user.logged }
                />
                
                <PrivateRoute 
                    path="/*"
                    element={ <DashboardRoutes /> }
                    isAutenticated={ user.logged }
                />
            </Routes>
        </div>
    )
}
