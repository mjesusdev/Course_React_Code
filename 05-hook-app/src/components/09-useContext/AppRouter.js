import React from 'react';
import {
    BrowserRouter as Router,
    Switch,
    Route,
    Redirect,
  } from 'react-router-dom';

import { NavBar } from './NavBar';

import { AboutScreen } from './AboutScreen';
import { LoginScreen } from './LoginScreen';
import { HomeScreen } from './HomeScreen';

export const AppRouter = () => {
    return (
        <Router>
            <div>
                
                <NavBar />

                <div className="container">
                    <Switch>
                        <Route exact path="/" component={ HomeScreen } />

                        <Route exact path="/about" component={ AboutScreen } />
                        <Route exact path="/login" component={ LoginScreen } />
                        
                        <Redirect to="/" />

                    </Switch>
                </div>
            </div>
        </Router>
    )
}
