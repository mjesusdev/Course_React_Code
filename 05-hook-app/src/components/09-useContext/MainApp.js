import React, { useState } from 'react'
import { AppRouter } from './AppRouter'
import { UserContext } from './UserContext'


export const MainApp = () => {

    const [user, setUser] = useState({});
    

    return (
        <UserContext.Provider value={{
            user,
            setUser
        }}>

            <AppRouter />

        </UserContext.Provider>
    )
}
