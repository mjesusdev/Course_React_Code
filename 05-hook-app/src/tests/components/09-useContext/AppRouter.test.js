import React from 'react';
import { mount } from 'enzyme';
import { AppRouter } from '../../../components/09-useContext/AppRouter';
import { UserContext } from '../../../components/09-useContext/UserContext';


describe('Pruebas en <AppRouter />', () => {
    
    const user = {
        id: 1,
        name: 'Fernando'
    }

    const wrapper = mount( 
        <UserContext.Provider value={{
            user
        }}>
            <AppRouter /> 
        </UserContext.Provider>
    );


    test('debe de mostrarse correctamente', () => {

        expect( wrapper ).toMatchSnapshot();
        
    })
    

})
