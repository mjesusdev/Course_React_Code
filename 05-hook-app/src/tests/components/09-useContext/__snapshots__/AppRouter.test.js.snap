// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <AppRouter /> debe de mostrarse correctamente 1`] = `
<div>
  <nav
    className="navbar navbar-expand-sm navbar-dark bg-dark"
  >
    <Link
      className="navbar-brand"
      to="/"
    >
      <LinkAnchor
        className="navbar-brand"
        href="/"
        navigate={[Function]}
      >
        <a
          className="navbar-brand"
          href="/"
          onClick={[Function]}
        >
           useContext 
        </a>
      </LinkAnchor>
    </Link>
    <div
      className="collapse navbar-collapse"
      id="navbarNavAltMarkup"
    >
      <div
        className="navbar-nav"
      >
        <NavLink
          activeClassName="active"
          className="nav-item nav-link"
          exact={true}
          to="/"
        >
          <Link
            aria-current="page"
            className="nav-item nav-link active"
            style={Object {}}
            to={
              Object {
                "hash": "",
                "pathname": "/",
                "search": "",
                "state": null,
              }
            }
          >
            <LinkAnchor
              aria-current="page"
              className="nav-item nav-link active"
              href="/"
              navigate={[Function]}
              style={Object {}}
            >
              <a
                aria-current="page"
                className="nav-item nav-link active"
                href="/"
                onClick={[Function]}
                style={Object {}}
              >
                 Home 
              </a>
            </LinkAnchor>
          </Link>
        </NavLink>
        <NavLink
          activeClassName="active"
          className="nav-item nav-link"
          exact={true}
          to="/about"
        >
          <Link
            aria-current={null}
            className="nav-item nav-link"
            to={
              Object {
                "hash": "",
                "pathname": "/about",
                "search": "",
                "state": null,
              }
            }
          >
            <LinkAnchor
              aria-current={null}
              className="nav-item nav-link"
              href="/about"
              navigate={[Function]}
            >
              <a
                aria-current={null}
                className="nav-item nav-link"
                href="/about"
                onClick={[Function]}
              >
                 About 
              </a>
            </LinkAnchor>
          </Link>
        </NavLink>
        <NavLink
          activeClassName="active"
          className="nav-item nav-link"
          exact={true}
          to="/login"
        >
          <Link
            aria-current={null}
            className="nav-item nav-link"
            to={
              Object {
                "hash": "",
                "pathname": "/login",
                "search": "",
                "state": null,
              }
            }
          >
            <LinkAnchor
              aria-current={null}
              className="nav-item nav-link"
              href="/login"
              navigate={[Function]}
            >
              <a
                aria-current={null}
                className="nav-item nav-link"
                href="/login"
                onClick={[Function]}
              >
                 Login
              </a>
            </LinkAnchor>
          </Link>
        </NavLink>
      </div>
    </div>
  </nav>
  <div
    className="container"
  >
    <div>
      <h1>
        HomeScreen
      </h1>
      <hr />
      <pre>
        {
   "id": 1,
   "name": "Fernando"
}
      </pre>
    </div>
  </div>
</div>
`;
